plugins/amazonq/shared/jetbrains-community/src/software/aws/toolkits/jetbrains/services/amazonq/project/ProjectContextProvider.kt

```diff
@@ -163,6 +171,14 @@ class ProjectContextProvider(val project: Project, private val encoderServer: En
         }
     }
 
+    fun queryInline(query: String, filePath: String): List<InlineBm25Chunk> {
+        val encrypted = encryptRequest(QueryInlineCompletionRequest(query, filePath))
+        return CompletableFuture.supplyAsync {
+            val r = sendMsgToLsp(LspMessage.QueryInlineCompletion, encrypted)
+            mapper.readValue<List<InlineBm25Chunk>>(r.responseBody)
+        }.get(50L, TimeUnit.MILLISECONDS)
```

Comment:
this will block the thread

Reply:
yea i know. 
But at the upstream caller https://github.com/aws/aws-toolkit-jetbrains/pull/4976/files#diff-1ffc2135fc40927c5c56f652967571eaa46b778e9a8496af29d6ed13ea1ee048R218-R242, this synchornous call is made within a suspend async coroutine, so it should be fine?

Reply:
i tried to make it suspend and use withTimeout, however it doesn't work because there is no suspension point within`sendMsgToLsp()` I guess.

![image](https://github.com/user-attachments/assets/2b653a9e-e24b-429a-956f-88525f686930)